tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java

```diff
@@ -59,22 +83,26 @@ public static void main(String[] args) throws Exception {
 
         // Configure bitcoinj to fetch only headers, not save them to disk, connect to a local fully synced/validated
         // node and to save block headers that are on interval boundaries, as long as they are <1 month old.
-        final BlockStore store = new MemoryBlockStore(PARAMS);
-        final BlockChain chain = new BlockChain(PARAMS, store);
-        final PeerGroup peerGroup = new PeerGroup(PARAMS, chain);
+        final BlockStore store = new MemoryBlockStore(params);
+        final BlockChain chain = new BlockChain(params, store);
+        final PeerGroup peerGroup = new PeerGroup(params, chain);
         final InetAddress peerAddress = InetAddress.getLocalHost();
         System.out.println("Connecting to " + peerAddress + "...");
         peerGroup.addAddress(peerAddress);
         long now = new Date().getTime() / 1000;
         peerGroup.setFastCatchupTimeSecs(now);
 
-        final long oneMonthAgo = now - (86400 * 30);
+        // Only go back 1 day for testnet, so we can quickly get past block storms
+        final int daysAgo = params.getId().equals(NetworkParameters.ID_MAINNET) ? 30 : 1;
```

Comment:
I'm not so sure if this is a good idea. The idea is checkpoints will never change. On testnet, larger forks have been seen than on mainnet.